Bigtable/src/Connection/Grpc.php

```diff
@@ -139,31 +140,87 @@ private function clusterObject(array $cluster)
      */
     public function getInstance(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $name = $this->pluck('name', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'getInstance'], [
+            $name,
+            $this->addResourcePrefixHeader($args, $name)
+        ]);
     }
 
     /**
      * @param array $args
      */
     public function listInstances(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $parent = $this->pluck('parent', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'listInstances'], [
+            $parent,
+            $this->addResourcePrefixHeader($args, $parent)
+        ]);
     }
 
     /**
      * @param array $args
      */
     public function updateInstance(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $name = $this->pluck('name', $args);
+        $displayName = $this->pluck('displayName', $args);
+        $type = $this->pluck('type', $args);
+        $labels = $this->pluck('labels', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'updateInstance'], [
+            $name,
+            $displayName,
+            $type,
+            $labels,
+            $this->addResourcePrefixHeader($args, $name)
+        ]);
     }
 
-        /**
+    /**
+     * @param array $args
+     */
+    public function partialUpdateInstance(array $args)
+    {
+        $parent = $this->pluck('parent', $args);
+        $instance = $this->pluck('instance', $args);
+        $updateMask = $this->pluck('updateMask', $args);
```

Comment:
In Cloud Spanner, I made the notion of an updateMask internal to the Connection method. The client provided whatever fields they wished to modify, and the connection method used that to create an update mask. If that's feasible in this case, it may make the client a bit simpler. What do you think?

https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/Spanner/src/Connection/Grpc.php#L242